From 3b8fbe38e20c758f8ecebb80cb00142fe84bc680 Mon Sep 17 00:00:00 2001
From: Hu hui <b29976@freescale.com>
Date: Wed, 20 Oct 2010 09:49:47 +0800
Subject: [PATCH] ENGR00132493 RAMDISK: doing write barrier will return -EIO

when doing write barrier, the data of the bio is NULL, it's OK for ramdisk
doing nothing, So the result of the bio must be success.

Signed-off-by: Hu Hui <b29976@freescale.com>
---
 drivers/block/brd.c |    7 ++++---
 1 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/block/brd.c b/drivers/block/brd.c
index f1bf79d..bcb50e8 100644
--- a/drivers/block/brd.c
+++ b/drivers/block/brd.c
@@ -333,13 +333,14 @@ static int brd_make_request(struct request_queue *q, struct bio *bio)
 	struct bio_vec *bvec;
 	sector_t sector;
 	int i;
-	int err = -EIO;
+	int err = 0;
 
 	sector = bio->bi_sector;
 	if (sector + (bio->bi_size >> SECTOR_SHIFT) >
-						get_capacity(bdev->bd_disk))
+						get_capacity(bdev->bd_disk)) {
+		err = -EIO;
 		goto out;
-
+	}
 	if (unlikely(bio_rw_flagged(bio, BIO_RW_DISCARD))) {
 		err = 0;
 		discard_from_brd(brd, sector, bio->bi_size);
-- 
1.5.4.4

